docs/tutorial.rst

```diff
@@ -869,6 +869,302 @@ of realistic size for real data.
+this scenario, let's simulate a selective sweep until it reaches an abitrary
+allele frequency.
+
+First, let's define a contig and a demographic model; here, we are simulating a
+small part of chromosome 2L of DroMel with a generic constant size demography.
+
+.. code-block:: python
+
+    import stdpopsim
+
+    species = stdpopsim.get_species("DroMel")
+    model = stdpopsim.PiecewiseConstantSize(100000)
+    samples = model.get_samples(100)
+    # Our contig will be fully neutral, with an exception to the mutation
+    # which will be subject to positive selection.
+    # TODO: remove the call to fully_neutral, once PR #959 goes in.
+    contig = species.get_contig("2L", length_multiplier=0.01)
+    contig.fully_neutral()
+
+Next, we need to define a mutation type which will be added to a randomly chosen
+chromosome in the population of our choice at a specific position in the contig.
+We must also decide the time the mutation will be added, when selection will
+start and at what frequency we want our selected mutation to be at the end of
+the simulation.
+
+Let's assume the mutation appeared 1000 generations ago, it has a positive
+effect on fitness (s=0.5). Also, we want the mutation to have reached a frequency
+of at least 0.8.
+
+.. note::
+
+    Note that because we are doing a forward-in-time simulation, you should be
+    careful with your conditioning. For example, even a strongly selected mutation
+    would not be able to reach 80% frequency in just a few generations.
+
+.. code-block:: python
+
+    # We also need to add a mutation type which will be added later in the simulaton
+    # and sweep.
+    contig.mutation_types.append(
+        stdpopsim.ext.MutationType(
+            distribution_type="f",
+            dominance_coeff=1.0,
+            distribution_args=[0.5],
+            convert_to_substitution=False,
+        )
+    )
+    mut_id = len(contig.mutation_types) - 1
+    # We will place our mutation in the middle of the contig.
+    coordinate = round(contig.recombination_map.sequence_length / 2)
+    # Time mutation will appear.
+    T_mut = 1000
+    extended_events = [
+        # Draw the mutation
+        stdpopsim.ext.DrawMutation(
+            time=T_mut,
+            mutation_type_id=mut_id,
+            population_id=0,
+            coordinate=coordinate,
+            # Save state before the mutation is introduced.
+            # This allows the simulation to restart from here if the mutation is lost.
+            save=True,
+        ),
+        # Condition on not loosing the mutation from the generation the mutation
+        # is added until the end. Note that our conditioning must start one
+        # generation after the mutation is placed. You cannot simply specify
+        # T_mut+1 if rescaling is present, otherwise the conditioning would start
+        # at the same generation when the mutation is placed.
+        stdpopsim.ext.ConditionOnAlleleFrequency(
```

Comment:
This conditioning is redundant - it can't have been lost if it's at frequency above 80% at the end. Is the purpose to restart the simulation earlier, if it's lost?

Reply:
yeah, the purpose is so that we don't wait until the end to check, instead we will be checking for presence every generation.